src/Microsoft.IdentityModel.KeyVaultExtensions/KeyVaultSignatureProvider.cs

```diff
@@ -41,11 +41,11 @@ public KeyVaultSignatureProvider(SecurityKey key, string algorithm, bool willCre
         /// <param name="algorithm">The signature algorithm to apply.</param>
         /// <param name="willCreateSignatures">Whether this <see cref="KeyVaultSignatureProvider"/> is required to create signatures then set this to true.</param>
         /// <param name="client">A mock <see cref="IKeyVaultClient"/> used for testing purposes.</param>
-        internal KeyVaultSignatureProvider(SecurityKey key, string algorithm, bool willCreateSignatures, IKeyVaultClient client)
+        internal KeyVaultSignatureProvider(SecurityKey key, string algorithm, bool willCreateSignatures, IKeyVaultClient? client)
             : base(key, algorithm)
         {
             _key = key as KeyVaultSecurityKey ?? throw LogHelper.LogArgumentNullException(nameof(key));
-            _client = client ?? new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(_key.Callback));
+            _client = client ?? new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(_key.Callback!));
```

Comment:
This seems to me like it could result in a NRE?

Could we handle that case? I am not sure what would be appropriate - does a KeyVaultClient without any callback make sense? We should not pass the callback if it is optional.

In case it is required I`d think that we should still throw ArgumentNullException on the Callback property then, because NRE are hard to debug with a misleading callstack. ANE would be a clear pointer to where the issue lay.

Reply:
Great point. In this case when we try to use the Callback from _key when callback is set on the the src/Microsoft.IdentityModel.KeyVaultExtensions/KeyVaultSecurityKey.cs we check it is null at the constructor 
        public KeyVaultSecurityKey(string keyIdentifier, AuthenticationCallback callback)
        {
            Callback = callback ?? throw LogHelper.LogArgumentNullException(nameof(callback));
            KeyId = keyIdentifier;
        }

Reply:
comment says this is a mock for test purposes and the method is internal.

Reply:
I understand, still making someone`s life harder.

Reply:
Given that this is not used other than for test purposes. I am resolving this as it is not high priority on overall work. Thanks for the feedback. It will be definitely good to keep it in mind for my next few PR's as a complete this task.